cli/command/service/update_test.go

```diff
@@ -587,6 +587,50 @@ func TestUpdateIsolationValid(t *testing.T) {
 	assert.Check(t, is.Equal(container.IsolationProcess, spec.TaskTemplate.ContainerSpec.Isolation))
 }
 
+// TestUpdateLimitsReservations tests that limits and reservations are updated,
+// and that values are not updated are not reset to their default value
+func TestUpdateLimitsReservations(t *testing.T) {
+	spec := swarm.ServiceSpec{
+		TaskTemplate: swarm.TaskSpec{
+			ContainerSpec: &swarm.ContainerSpec{},
+			Resources: &swarm.ResourceRequirements{
+				Limits: &swarm.Resources{
+					NanoCPUs:    1000000000,
+					MemoryBytes: 104857600,
+				},
+				Reservations: &swarm.Resources{
+					NanoCPUs:    1000000000,
+					MemoryBytes: 104857600,
+				},
+			},
+		},
+	}
+
+	flags := newUpdateCommand(nil).Flags()
+	err := flags.Set(flagLimitCPU, "2")
+	assert.NilError(t, err)
+	err = flags.Set(flagReserveCPU, "2")
+	assert.NilError(t, err)
+	err = updateService(context.Background(), nil, flags, &spec)
+	assert.NilError(t, err)
+	assert.Check(t, is.Equal(spec.TaskTemplate.Resources.Limits.NanoCPUs, int64(2000000000)))
+	assert.Check(t, is.Equal(spec.TaskTemplate.Resources.Limits.MemoryBytes, int64(104857600)))
+	assert.Check(t, is.Equal(spec.TaskTemplate.Resources.Reservations.NanoCPUs, int64(2000000000)))
+	assert.Check(t, is.Equal(spec.TaskTemplate.Resources.Reservations.MemoryBytes, int64(104857600)))
+
+	flags = newUpdateCommand(nil).Flags()
```

Comment:
nit: split it into 2 subtests?

Reply:
Yep, in case of failure it points directly to the right sub-test. But it's only a small nit, no need to update it 😄